packages/dds/tree/docs/wip/schema-evolution.md

```diff
@@ -0,0 +1,583 @@
+   A new version of the application is deployed which will upgrade the document to stored schema B when it first loads (or first writes to) a document using stored schema A.
+
+   If every client is guaranteed to have write permissions, then the developer can also remove view schema A and the code written against it in the same deployment.
+   This is because any client that encounters a document with stored schema A can upgrade it to stored schema B before interacting with it.
+   However, if the client is a read-only client, then it can't do this, and it needs to retain the code to read view schema A in case it opens a document with stored schema A.
+   This caveat is made more manageable by [adapters](#adapters), introduced later in this document.
+
+     > To minimize the time between client saturation and document upgrades, the ability to upgrade the document can be made part of the deployment in step 1 but "shipped dark" and remotely enabled in this step via a feature flight.
+
+4. Wait until all documents have upgraded to the new stored schema.
+   After this **document saturation** has been achieved, the application can do another deployment, removing view schema A and all of its related code, if it was not able to do so in step 3 (because some of its clients are read-only).
+   It's important that this happens after document saturation, or else there will be read-only clients that fail to open a document with stored schema A because they only know about view schema B.
+
+   > In some environments, there's no guarantee that document saturation will ever occur - and this step is never completed.
+     Old documents can hang around forever if they remain unopened, but must still be supported in the event that someday they are needed.
+     The code that reads view schema A must be maintained forever.
+     Fortunately, this headache can be somewhat mitigated via the use of [adapters](#adapters).
+
+5. Eventually, all clients will have updated to the latest deployment, and all traces of schema A will be gone in both the documents and application source code.
+
+```mermaid
+gantt
+dateFormat D
+axisFormat Step %e
+tickInterval 1day
+Maintain view schema A in source code (read/write client) : 1, 2d
+Maintain view schema A in source code (read-only client)  : 1, 3d
+Documents contain stored schema A                         : 1, 3d
+Maintain view schema B in source code                     : 2, 3d
+Documents contain stored schema B                         : 3, 2d
+```
+
+Different applications with different requirements may tweak this process in various ways, but in general doing a staged rollout is a cumbersome endeavor.
+It takes time, it requires code bifurcation/duplication, and it's complicated to understand and coordinate.
+The next sections will explore ways to reduce the amount of effort a developer must take to change their schema.
+
+### Adapters
```

Comment:
I assume adapters do not exist today (i.e. it's concept of the future).
That said even if it's the case, it would be nice to see some example pseudo-code.
Som readers would find reading code way more rewarding than reading English :)